src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs

```diff
@@ -146,5 +161,13 @@ public async Task<bool> InstanceSchemaRecordExistsAsync(CancellationToken cancel
             using var command = new SqlCommand(procedureQuery, connection);
             return (int)await command.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false) != 0;
         }
+
+        private ServerConnection GetServerConnectionWithTimeout(SqlConnection sqlConnection)
+        {
+            var serverConnection = new ServerConnection(sqlConnection);
+            serverConnection.StatementTimeout = _sqlServerDataStoreConfiguration.StatementTimeoutInSecs;
```

Comment:
Just for my understanding, Is serverConnection.StatementTimeout same as server.ConnectionContext.StatementTimeout (updating it before line 60)? For my test, I used "server.ConnectionContext.StatementTimeout".

Reply:
Yes, statementTimeout is property on serverConnection only. When we set StatementTimeout to server.ConnectionContext, its also adding timeout to serverConnection itself.